File: leet/two_sum/two_sum.dart
class Solution {
  List<int> twoSum(List<int> nums, int target) {
    Map<int, int> map = {};
    for (var i = 0; i < nums.length; i++) {
      if (map.containsKey(target - nums[i])) {
        return [i, map[target - nums[i]]!];
      }
      map[nums[i]] = i;
    }
    return [];
  }
}

File: leet/two_sum/two_sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (nums, target) {
    var map = new Map();
    for (let index = 0; index < nums.length; index++) {
        if (map.has(target - nums[index])) {
            return [index, map.get(target - nums[index])];
        }
        map.set(nums[index], index);
    }
    return [];
};